pkg/provider/azure_loadbalancer.go

```diff
@@ -1356,20 +1364,30 @@ func (az *Cloud) isFrontendIPChanged(clusterName string, config network.Frontend
 	if !strings.EqualFold(pointer.StringDeref(config.Name, ""), lbFrontendIPConfigName) {
 		return false, nil
 	}
-	isIPv6 := utilnet.IsIPv6String(service.Spec.ClusterIP)
-	loadBalancerIP := getServiceLoadBalancerIP(service, isIPv6)
 	isInternal := requiresInternalLoadBalancer(service)
+	isIPv6, err := az.isFIPIPv6(&config, isInternal)
+	if err != nil {
+		return false, err
+	}
+	loadBalancerIP := getServiceLoadBalancerIP(service, isIPv6)
 	if isInternal {
 		// Judge subnet
-		subnetName := subnet(service)
+		subnetName := getInternalSubnet(service)
 		if subnetName != nil {
-			subnet, existsSubnet, err := az.getSubnet(az.VnetName, *subnetName)
-			if err != nil {
-				return false, err
+			if subnet == nil {
+				return false, fmt.Errorf("subnet is nil, should not happen")
 			}
-			if !existsSubnet {
-				return false, fmt.Errorf("failed to get subnet")
+			if subnet.ID == nil {
```

Comment:
when can this happen? when subnet is not nil but subnet.ID is nil?

Reply:
I think first subnet is always not nil. Then, before subnet is `az.getSubnet()` for the first time, its ID is nil.